docs/guide/templates.md

```diff
@@ -131,3 +131,73 @@ class MyComponent < ViewComponent::Base
   strip_trailing_whitespace(false)
 end
 ```
+
+## Sub-templates
+
+Since 2.73.0
+{: .label }
+
+Experimental
+{: .label .label-yellow }
+
+ViewComponents can render sub-templates defined in the sidecar directory:
+
+```text
+app/components
+├── ...
+├── test_component.rb
+├── test_component
+|   ├── list.html.erb
+|   └── summary.html.erb
+├── ...
+```
+
+Templates are compiled to methods in the format `render_#{template_basename}_template`, which can then be called in the component.
+
+```ruby
+class TestComponent < ViewComponent::Base
+  def initialize(mode:)
+    @mode = mode
+  end
+
+  def call
+    case @mode
+    when :list
+      render_list_template
+    when :summary
+      render_summary_template
+    end
+  end
+end
+```
+
+To define which parameters a template accepts, use the
+[Rails Strict Locals](https://edgeguides.rubyonrails.org/action_view_overview.html#strict-locals) syntax.
+Unlike Rails, a missing `locals` declaration means the template takes no arguments.
+This option only affects sub-templates.
+
+```ruby
+# test_component.rb
+class TestComponent < ViewComponent::Base
+  def initialize(mode:)
+    @mode = mode
+  end
+
+  def call
+    case @mode
+    when :list
+      render_list_template multiple: false
+    when :multilist
+      render_list_template multiple: true
+    when :summary
+      render_summary_template
+    end
+  end
+end
+```
+
+```erb
+<%# list.html.erb %>
+<%# locals: (multiple:) %->
+The parameter is <%= multiple %>
+```
```

Comment:
Might we put this example block above the Ruby? It's the more interesting bit IMO